op-program/client/l1/oracle.go

```diff
@@ -86,3 +91,31 @@ func (p *PreimageOracle) ReceiptsByBlockHash(blockHash common.Hash) (eth.BlockIn
 
 	return info, receipts
 }
+
+func (p *PreimageOracle) GetBlob(ref eth.L1BlockRef, blobHash eth.IndexedBlobHash) *eth.Blob {
+	indexBytes := make([]byte, 32)
+	binary.BigEndian.PutUint64(indexBytes, blobHash.Index)
+
+	// Hint for the commitment, the preimage of the blob versioned hash.
+	p.hint.Hint(Sha2Hint(blobHash.Hash))
```

Comment:
We should include the `ref.timestamp` in the hint, as well as the index, that can then be translated into `slot`  ( `(time - genesis_time)/12 = slot` in the beacon-chain), to then load the beacon-block side-cars that match this L1 block. And then filter down by the index that comes with the indexed-hash.

So same thing as the below hint. Might also just be worth writing a single hint, and then as host write both the sha-2 preimage (the kzg commitment bytes48) into the store as well as the blob points.